ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -211,6 +244,35 @@ public void readInto(ByteBuffer buffer, long position) throws IOException {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   * <p/>
+   * @param byteArray The buffer into which the data needs to be written.
+   * @param position The source's position to start the read.
+   * @param length The length of data in bytes to read.
+   * @throws IOException if data could not be written to the file because of I/O errors.
+   * @throws IndexOutOfBoundsException if {@code position} < header size or >= {@link #sizeInBytes()} or if
+   * {@code buffer} size is greater than the data available for read.
+   */
+  public void readIntoDirectly(byte[] byteArray, long position, int length) throws IOException {
+    long sizeInBytes = sizeInBytes();
```

Comment:
have a shared function do all the prechecks?

Reply:
Fixed.